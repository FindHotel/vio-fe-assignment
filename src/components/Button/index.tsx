import React, { ReactElement, FunctionComponent } from "react";

import { StyledButton } from "./styles";

interface ButtonProps {
  theme?: string;
  onClick: () => void;
  outline?: boolean;
  block?: boolean;
}

const Button: FunctionComponent<ButtonProps> = ({
  theme = "primary",
  children,
  onClick,
  outline = false,
  block = true,
}): ReactElement => {
  return (
    <StyledButton
      onClick={onClick}
      theme={theme}
      outline={outline}
      block={block}
    >
      {children}
    </StyledButton>
  );
};

export default Button;
