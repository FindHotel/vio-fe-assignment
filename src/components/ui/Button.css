.button {
    background: #f7fbff;
    border: 1px solid #bfdaf9;
    color: #0071f3;
    font-weight: 600;
    padding: 0 12px;
    border-radius: 6px;
    cursor: pointer;
    height: 40px;
}

.button:disabled {
    opacity: .4;
    cursor: default;
}

.button_contained {
    background: #0071f3;
    box-shadow: 0 0 1px rgba(68, 80, 95, 0.08), 0 1px 4px rgba(68, 80, 95, 0.32);
    color: #fff;
}

.button_content {
    display: flex;
    align-items: center;
    justify-content: center;
}

.button_icon {
    display: flex;
    align-items: center;
}

.button_icon_margin {
    margin-right: 10px;
}

.button_fullwidth {
    width: 100%;
}

.button_transparent {
    background: transparent;
    border: 0;
}