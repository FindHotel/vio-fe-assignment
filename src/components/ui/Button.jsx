import React from 'react';
import cn from 'classnames';

import './Button.css';

const Button = ({
  children,
  disabled,
  icon,
  onClick,
  fullwidth,
  contained,
  className,
  transparent,
}) => (
  <button
    className={cn(
      'button',
      { button_contained: contained, button_fullwidth: fullwidth, button_transparent: transparent }, className,
    )}
    type="button"
    disabled={disabled}
    onClick={onClick}
  >
    <span className="button_content">
      {icon && <span className={cn('button_icon', { button_icon_margin: children })}>{icon}</span>}
      {children}
    </span>
  </button>
);

export default Button;
