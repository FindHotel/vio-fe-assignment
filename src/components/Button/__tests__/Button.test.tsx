import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import Button from "..";

describe("<Button />", () => {
  it("should render properly", () => {
    const { container } = render(
      <Button onClick={() => {}}>Button content</Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should trigger onClick prop", () => {
    const mockedOnClick = jest.fn();
    const buttonText = "Button Content";

    render(<Button onClick={mockedOnClick}>{buttonText}</Button>);

    fireEvent.click(screen.queryByText(buttonText));

    expect(mockedOnClick).toHaveBeenCalled();
  });
});
