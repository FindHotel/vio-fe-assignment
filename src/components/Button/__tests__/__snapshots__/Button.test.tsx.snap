// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
<button
  class="sc-bdvvaa ktiVbg"
>
  Button content
</button>
`;
